index.js

```diff
@@ -127,28 +127,35 @@ class SimpleKeyring extends EventEmitter {
     return Promise.resolve(sig)
   }
 
-  getPrivateKeyFor (address, opts = {}) {
+  getPrivateKeyFor (address) {
     if (!address) {
       throw new Error('Must specify address.');
     }
-    const wallet = this._getWalletForAccount(address, opts)
+    const wallet = this._getWalletForAccount(address)
     const privKey = ethUtil.toBuffer(wallet.getPrivateKey())
     return privKey;
   }
 
-  // returns an address specific to an app
-  getAppKeyAddress (address, origin) {
-    return new Promise((resolve, reject) => {
-      try {
-        const wallet = this._getWalletForAccount(address, {
-          withAppKeyOrigin: origin,
-        })
-        const appKeyAddress = sigUtil.normalize(wallet.getAddress().toString('hex'))
-        return resolve(appKeyAddress)
-      } catch (e) {
-        return reject(e)
-      }
-    })
+  getAppKeyring (address, origin) {
+    if (
+      !origin ||
+      typeof origin !== 'string'
+    ) {
+      throw new Error(`'origin' must be a non-empty string`)
+    }
+
+    const wallet = this._getWalletForAccount(address)
+    const privKey = wallet.getPrivateKey()
+    const appKeyOriginBuffer = Buffer.from(origin, 'utf8')
+    const appKeyBuffer = Buffer.concat([privKey, appKeyOriginBuffer])
+    const appKeyPrivKey = ethUtil.keccak(appKeyBuffer, 256)
+    return new SimpleKeyring([appKeyPrivKey.toString('hex')])
+  }
+
+  async getAppKeyAddress (address, origin) {
+    const keyring = this.getAppKeyring(address, origin)
+    const accounts = await keyring.getAccounts()
+    return accounts[0]
   }
 
   // exportAccount should return a hex-encoded private key:
```

Comment:
There seems to be a tiny problem with newly introduced exportAccount opts:

https://github.com/MetaMask/eth-simple-keyring/commit/91645ed798ccc7627ad39948a157e9bd738bbcd3#diff-168726dbe96b3ce427e7fedce31bb0bcR146-R147

Reply:
Ah right, that should be removed. Thanks!